libs/api/domains/assets/src/lib/api-domains-assets.service.ts

```diff
@@ -242,4 +242,22 @@ export class AssetsXRoadService {
 
     throw new Error('Could not fetch fasteignir')
   }
+
+  async getRealEstatesWithDetail(
+    auth: User,
+    cursor?: string | null,
+  ): Promise<any | null> {
+    const getRealEstatesRes = await this.getRealEstates(auth, cursor)
+    if (getRealEstatesRes && getRealEstatesRes.properties) {
+      return {
+        paging: getRealEstatesRes.paging,
+        properties: getRealEstatesRes.properties.map(
```

Comment:
This will result in Promise array for `properties` you need to use Promise.all to resolve
```suggestion
        properties: await Promise.all(getRealEstatesRes.properties.map(
```

If the properties list can be large you might need to look into batching this.